deploy/deployUtils.ts

```diff
@@ -765,3 +770,166 @@ export async function stealFundsFromWhales(
     )
   }
 }
+
+export async function deployPermissionlessPoolComponents(
+  hre: HardhatRuntimeEnvironment,
+  metaswapDeposit: Address,
+) {
+  const { deployments, getNamedAccounts, getChainId, ethers } = hre
+  const { deploy, get, getOrNull, execute } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  const permissionlessSwap = await getOrNull("PermissionlessSwap")
+  const permissionlessMetaSwap = await getOrNull("PermissionlessMetaSwap")
+  const permissionlessDeployer = await getOrNull("PermissionlessDeployer")
+  const masterRegistry: MasterRegistry = await ethers.getContract(
+    "MasterRegistry",
+    deployer,
+  )
+  const masterRegistryAddress = masterRegistry.address
+  const swapUtilsAddress = (await get("SwapUtils")).address
+  const amplificationUtilsAddress = (await get("AmplificationUtils")).address
+
+  // see if master registry has a fee collector set
+  const feeCollectorName = ethers.utils.formatBytes32String("FeeCollector")
+
+  try {
+    await masterRegistry.resolveNameToLatestAddress(feeCollectorName)
+  } catch (error) {
+    console.log("No fee collector set, setting now")
+    // setting as the deployer for now as no multisig is available on this network
+    await masterRegistry.addRegistry(feeCollectorName, deployer)
+    console.log("Successfully set fee collector")
+  }
+
+  // deploy PermissionlessSwap if needed
+  if (permissionlessSwap == null) {
+    console.log("PermissionlessSwap not found, deploying")
+    const permissionlessSwapDeployment = await deploy("PermissionlessSwap", {
+      from: deployer,
+      log: true,
+      skipIfAlreadyDeployed: true,
+      args: [masterRegistryAddress],
+      libraries: {
+        SwapUtils: swapUtilsAddress,
+        AmplificationUtils: amplificationUtilsAddress,
+      },
+    })
+    console.log(
+      `PermissionlessSwap deployed at ${permissionlessSwapDeployment.address}`,
+    )
```

Comment:
This will produce two logs. One from setting `log : true` on the deploy settings and one from `console.log` line.

You can remove the second log since its redundant.